src/context/NotificationsContext/NotificationsContextProvider.tsx

```diff
@@ -19,6 +20,22 @@ const NotificationsContextProvider = ({
   const [createUserNotificationsData] =
     useCreateUserNotificationsDataMutation();
 
+  const {
```

Comment:
we've already discussed this on Discord, but would it make sense to have this separate from the global metadata notifications context? or just calling this hook directly in components that render a list of notifications?

Reply:
I want to avoid specific uses of any magicbell code as much as possible outside of this context (some places it is unavoidable unfortunately) so that if we ever decided to switch to a new provider, most of the logic is contained in this one place. I've tried to follow a similar pattern in the block ingestor too. We do need to separate this context into two though to fix https://github.com/JoinColony/colonyCDapp/issues/3168 but lets handle it in that PR, we can pair on it if you want too!